src/main/java/rx/internal/operators/OperatorMerge.java

```diff
@@ -545,16 +548,23 @@ public void request(long n) {
             if (n == Long.MAX_VALUE) {
                 requested = Long.MAX_VALUE;
             } else {
-                BackpressureUtils.getAndAddRequest(REQUESTED, this, n);
+                final long count = BackpressureUtils.getAndAddRequest(REQUESTED, this, n);
                 if (ms.drainQueuesIfNeeded()) {
                     boolean sendComplete = false;
+                    boolean requestMore = false;
                     synchronized (ms) {
                         if (ms.wip == 0 && ms.scalarValueQueue != null && ms.scalarValueQueue.isEmpty()) {
                             sendComplete = true;
                         }
+
+                        if (count > 0 && ms.wip == 0 && ms.scalarValueQueue == null) {
```

Comment:
This is here to allow requesting from the upstream producer if there are no pending requests. Without these the request for more will never reach the source observable.